src/core/Carousel/CarouselDotsList.tsx

```diff
@@ -0,0 +1,157 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { CarouselContext } from './CarouselContext';
+import { useMergedRefs, useResizeObserver, useTheme } from '../utils';
+import { CarouselDot } from './CarouselDot';
+import '@itwin/itwinui-css/css/carousel.css';
+
+type CarouselDotsListProps = {
+  /** Number of total dots/slides in the carousel. Will be inferred from Carousel context. */
+  length?: number;
+  /** Maximum number of dots to show in the viewport. */
+  maxCount?: number;
```

Comment:
This `maxCount `prop is in a weird situation now after adding the resizeObserver logic. Maybe I should just remove this prop? If not, we will have to do some hacks like this:

```tsx
style={{ maxWidth: maxCount * dotSize }}
```

Reply:
Removed maxCount 🤷